lib/rubocop/cop/lint/overwrite_by_rescue.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # Checks for overwriting an exception with an exception result by use `rescue =>`.
```

Comment:
I'd expand on the documentation here and add some explanation that people wrote a `rescue` modifier form instead of rescue that actually catches exception, which they probably meant to write.

Reply:
Documentation has been expanded to include additional explanations.